builders/nuget.go

```diff
@@ -0,0 +1,231 @@
+		if err != nil {
+			return fmt.Errorf("could not run `nuget install`: %s", err.Error())
+		}
+	} else {
+		return errors.New("No tools installed in local environment for NuGet build")
+	}
+
+	nugetLogger.Debug("Done running NuGet build.")
+	return nil
+}
+
+// Analyze parses the output of NuGet lockfiles and falls back to parsing the packages folder
+func (builder *NuGetBuilder) Analyze(m module.Module, allowUnresolved bool) ([]module.Dependency, error) {
+	nugetLogger.Debugf("Running NuGet analysis: %#v %#v", m, allowUnresolved)
+
+	deps := []module.Dependency{}
+
+	// Find and parse a lockfile
+	lockFilePath, err := resolveProjectLockfile(m.Dir)
+	if err == nil {
+		parseProjectLockfile(filepath.Join(m.Dir, lockFilePath))
+	} else {
+		// Fallback to parsing the packages directory
+		packagesDir, err := resolveNugetPackagesDir(m.Dir)
+
+		nugetLogger.Debugf("Parsing packages directory: %s", packagesDir)
+		if exists, err := hasFile(packagesDir); err != nil || !exists {
+			return nil, fmt.Errorf("Unable to verify packages directory: %s", packagesDir)
+		}
+
+		packagePaths, err := ioutil.ReadDir(packagesDir)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, f := range packagePaths {
+			packageNameRe := regexp.MustCompile(`(([A-z]+\.?)+)\.(([0-9]+\.)+[\w-]+)`)
+			match := packageNameRe.FindStringSubmatch(f.Name())
+			nugetLogger.Debugf("%s, %V", len(match), match)
+			if len(match) == 5 {
+				deps = append(deps, module.Dependency(NuGetPackage{
+					Name:    match[1],
+					Version: match[3],
+				}))
+			}
+		}
+	}
+
+	// TODO: filter out system deps
+
+	nugetLogger.Debugf("Done running NuGet analysis: %#v", deps)
+	return deps, nil
+}
+
+// IsBuilt checks the existance of a project.lock.json or a packages directory
+func (builder *NuGetBuilder) IsBuilt(m module.Module, allowUnresolved bool) (bool, error) {
+	if allowUnresolved {
+		return true, nil
+	}
+
+	nugetLogger.Debug("Checking NuGet module directory for a project lockfile")
+	if _, err := resolveProjectLockfile(m.Dir); err != nil {
+		nugetLogger.Debug("Checking NuGet packages directory for existence")
+
+		packagesDir, _ := resolveNugetPackagesDir(m.Dir)
+		return hasFile(packagesDir)
+	}
+
+	return true, nil
+}
+
+// IsModule is not implemented
+func (builder *NuGetBuilder) IsModule(target string) (bool, error) {
+	return false, errors.New("IsModule is not implemented for NuGetBuilder")
+}
+
+// parseProjectLockfile parses the lockfile
+func parseProjectLockfile(filename string) {
+	// TODO: actually parse the lockfiles
+	return
+}
+
+// resolveNugetPackagesDir parses a NuGet module config and resolves it to an existing package directory
+func resolveNugetPackagesDir(dir string) (string, error) {
+	packagesDir := filepath.Join(dir, "packages")
+	return packagesDir, fmt.Errorf("unable to resolve NuGet packages directory: %s", "Not Implemented.")
+}
+
+func resolveProjectLockfile(dir string) (string, error) {
```

Comment:
Would prefer `resolveNugetProjectLockfile` because these names are visible to all of `builders`.